pkg/driver/node.go

```diff
@@ -73,12 +93,41 @@ func (d *Driver) NodePublishVolume(ctx context.Context, req *csi.NodePublishVolu
 	// TODO when CreateVolume is implemented, it must use the same key names
 	subpath := "/"
 	encryptInTransit := true
+	iamAuth := false
 	volContext := req.GetVolumeContext()
+
 	for k, v := range volContext {
+
 		switch strings.ToLower(k) {
+		case "csi.storage.k8s.io/pod.name":
+			continue
+		case "csi.storage.k8s.io/pod.namespace":
+			continue
+		case "csi.storage.k8s.io/pod.uid":
+			continue
+		case "csi.storage.k8s.io/serviceaccount.name":
+			continue
+		case "csi.storage.k8s.io/serviceaccount.tokens":
+			//TODO get the token in a cleaner way, not like this.
+			result := make(map[string]interface{})
+			json.Unmarshal([]byte(volContext["csi.storage.k8s.io/serviceAccount.tokens"]), &result)
+			tokenInfo := fmt.Sprintf("%v", result["sts.amazonaws.com"].(map[string]interface{})["token"])
+
```

Comment:
Could result be a `map[string]map[string]string` here? Not the prettiest data type but it avoids the use of `interface{}`. Further I wonder if there's a JWT library that would make this extraction a little bit easier?

Reply:
The tokens are returned as an array of json string, not sure the JWT library would help here.